### PR TITLE
fix(questionnaires): three live defects from the 2026-09-10 tech-debt sweep

### LINKED ISSUE #956: Tech-debt sweep 2026-09-10: three live defects (LLM null guidelines crash, admission staleness deadlock, MC-option response drops id)
Three correctness defects surfaced by the 2026-09-10 tech-debt sweep, independently re-verified against `main` (83c0a259). Tracked together because each is a one-line root cause with a clear regression test.

## 1. Automatic free-text evaluation crashes when a question has no per-question LLM guidelines

- `FreeTextQuestion.llm_guidelines` is `TextField(null=True)` and the create schema defaults it to `None`.
- `SubmissionEvaluator._evaluate_ft_answers_in_batch` feeds that into `AnswerToEvaluate.guidelines: str` (non-optional pydantic field) → `ValidationError`.
- Reproduced: `AnswerToEvaluate(..., guidelines=None)` raises `string_type` validation error.
- `evaluate()` calls the free-text batch unconditionally, so any submission with an applicable free-text answer on a guideline-less question fails; the Celery task re-raises and the submission stays unevaluated.
- The only test fixture hardcodes guidelines, which is why this never surfaced.

**Fix:** make `guidelines: str | None = None`, tolerate `None` in the backends' escape path, add a regression test with `llm_guidelines=None`.

## 2. Admission-questionnaire staleness deadlock

- `EventQuestionnaireGate._is_submission_expired` treats an APPROVED submission older than `max_submission_age` as missing and answers `COMPLETE_QUESTIONNAIRE`.
- `_validate_admission_resubmission` has no staleness branch and 400s "already approved" on any APPROVED evaluation.
- The membership twin (`membership_questionnaire_service._approval_is_stale`) handles this correctly, proving the drift.
- With `per_event=True` (or a same-event approval) the user is told to resubmit and then refused: an unrecoverable loop with no organizer-visible cause.

**Fix:** add the staleness branch to `_validate_admission_resubmission`, sharing the helper with the membership service. Extracting the full retake/staleness policy (report finding 12) stays a follow-up.

## 3. MC-option create/update endpoints declare a response schema without `id`

- All four routes (`polls` + `questionnaires` controllers, POST + PUT) use `MultipleChoiceOptionUpdateSchema` as `response=`; ninja strips `id`.
- `.artifacts/openapi.json` confirms the `$ref` points at the update schema.
- Frontend `poll-api-sync-questions.ts` and `questionnaire-api-sync-questions.ts` cast the response to read `id`, get `undefined`, and silently skip conditional-question/section sync for every newly created option.
- `test_controller_drift.py` compares only path and method, so the two controllers could not have drifted apart here, but both are wrong.

**Fix:** switch the four `response=` declarations to `MultipleChoiceOptionResponseSchema`, regenerate the OpenAPI artifact, extend the drift test to compare response schemas. Frontend follow-up: `pnpm generate:api` and drop the cast.

### PR DESCRIPTION
**Pull request by biagiodistefano**
Closes #956

Independently re-verified findings 1–3 of the 2026-09-10 tech-debt report against `main`, then fixed each test-first.

## Changes

**1. Null per-question LLM guidelines crashed automatic evaluation**
- `AnswerToEvaluate.guidelines` → `str | None = None`; the OpenAI backend only escapes it when present (the per-question template already guards with `{% if response.guidelines %}`).
- Regression: `test_free_text_evaluation_tolerates_question_without_llm_guidelines` (reproduced the pydantic `string_type` error before the fix).

**2. Admission-questionnaire staleness deadlock**
- `_validate_admission_resubmission` gains the same stale-approval branch the membership path has. `_approval_is_stale` is renamed `approval_is_stale` and imported, not copied.
- Regression: `TestAdmissionStaleApproval` (stale → allowed, fresh → still 400 "already approved").
- The wider retake/staleness policy extraction (report finding 12) is intentionally out of scope.

**3. MC-option create/update responses dropped `id`**
- All four routes (`polls` + `questionnaires`, POST + PUT) now declare `MultipleChoiceOptionResponseSchema`.
- `test_controller_drift.py` gains `test_polls_and_org_questionnaire_response_schemas_in_sync` so a one-sided fix can't land again.
- Existing create-option tests now assert `id` in the body. `.artifacts/openapi.json` regenerated (gitignored).

## Verification
- `make check` green.
- 196 passed across the questionnaire/poll/event-questionnaire test files plus `test_error_response_contracts.py`.

## Follow-up (frontend)
`pnpm generate:api`, then drop the `(response.data as { id?: string })` casts in `poll-api-sync-questions.ts` and `questionnaire-api-sync-questions.ts`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Approved questionnaire submissions can be resubmitted after their configured approval period expires.
  * Multiple-choice option responses now consistently include the created option’s ID.
  * Automatic evaluation supports free-text questions without item-specific guidelines.

* **Bug Fixes**
  * Freshly approved submissions remain protected from duplicate resubmission.
  * Improved handling of optional evaluation guidelines.
  * Synchronized response formats across questionnaire and poll question endpoints.

* **Tests**
  * Added coverage for stale approvals, response IDs, endpoint consistency, and guideline-free evaluations.

### CODE REVIEW

File: src/questionnaires/tests/test_evaluator.py [hunk 1]
```diff
@@ -494,3 +494,25 @@ def test_non_applicable_questions_not_counted_in_max_points(
...
+    submitted_submission.questionnaire.evaluation_mode = Questionnaire.QuestionnaireEvaluationMode.AUTOMATIC
+    submitted_submission.questionnaire.save()
+    question = FreeTextQuestion.objects.create(
+        questionnaire=submitted_submission.questionnaire, question="Why?", order=1, llm_guidelines=None
+    )
+    FreeTextAnswer.objects.create(submission=submitted_submission, question=question, answer="Because it is good.")
+
+    evaluation = SubmissionEvaluator(submission=submitted_submission, llm_evaluator=mock_evaluator).evaluate()
+
+    assert evaluation.evaluation_data.ft_points_scored == Decimal("1.0")
```
[the remark this answers is not in the document]
Comment (biagiodistefano) [reply src/questionnaires/tests/test_evaluator.py:518]: Added `test_free_text_rejection_without_llm_guidelines_scores_zero` in 7725c13480b54467d304c682d826af2fc6050102: same guideline-less question, an answer the mock evaluator rejects, asserting zero free-text points and a REJECTED status.

On the Linked Issues warning: `.artifacts/openapi.json` is gitignored (built during the Docker image build), so its regeneration cannot show up in the diff. It was regenerated locally and all four MC-option routes now reference `MultipleChoiceOptionResponseSchema`.
